autosklearn/evaluation/abstract_evaluator.py

```diff
@@ -410,7 +410,8 @@ def finish_up(
 
         additional_run_info = {} if additional_run_info is None else additional_run_info
         for metric in self.scoring_functions:
-            additional_run_info[metric.name] = loss_[metric.name]
+            if metric.name in loss_:
+                additional_run_info[metric.name] = loss_[metric.name]
```

Comment:
Yes, correct. This can happen if the user requests a metric that cannot be computed on the current task, for example `precision` on a multiclass problem.